### PR TITLE
Clarifying usage of environment variables on the command line

### LINKED ISSUE #1458: Using environment variables to enable security and privacy: Small improvement needed for instructions
## Description

In the document "Setting up a Network for Development", under "Start up the first validating peer" heading, it is shown how to enable security and privacy using environment variables (without editing the core.yaml file).

However, in the subsequent sections such as "Starting up the second validating peer", "Deploy, Invoke and Query a Chaincode", the environment variables are omitted. This leads to the error 'Error: Error building chaincode: rpc error: code = 2 desc = "Registration to the Membership Service required."' while deploying a chaincode.

So please update the instructions as mentioned here,
https://github.com/hyperledger/fabric/issues/1185#issuecomment-214878753
## Describe How to Reproduce

https://github.com/hyperledger/fabric/blob/master/docs/dev-setup/devnet-setup.md

### PR DESCRIPTION
**Pull request by angrbrd**
## Description

These changes are documentation changes only. They are put in place to address several concerns with confusion on the usage of environment variables on the command line.
## Motivation and Context

The change adds modifications to existing examples of command line usage for deploy, invoke, query, and starting up docker containers where environment variables are used.

Fixes #1458
## How Has This Been Tested?

Documentation changes only. No tests required.
## Checklist:
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Fixes #1458

Signed-off-by: Anna D Derbakova adderbak@us.ibm.com

### CODE REVIEW

File: docs/API/SandboxSetup.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ To set up the local development environment with security enabled, you must firs
...
+**Note:** To run with security enabled, first modify the [core.yaml](https://github.com/hyperledger/fabric/blob/master/peer/core.yaml) configuration file to set the <b>security.enabled</b> value to 'true' before building the peer executable. Alternatively, you can enable security by running the peer with environment variable CORE_SECURITY_ENABLED=true. To enable privacy and confidentiality of transactions (requires security to also be enabled), modify the [core.yaml](https://github.com/hyperledger/fabric/blob/master/peer/core.yaml) configuration file to set the <b>security.privacy</b> value to 'true' as well. Alternatively, you can enable privacy by running the peer with environment variable CORE_SECURITY_PRIVACY=true. If you are enabling security and privacy on the peer process with environment variables, it is important to include these environment variables in the command when executing all subsequent peer operations (e.g. deploy, invoke, or query).
```
Comment (srderson) [inline docs/API/SandboxSetup.md]: Could you use ` around
`security.enabled`
`CORE_SECURITY_ENABLED=true`
`true`

File: docs/API/SandboxSetup.md [hunk 2]
```diff
@@ -152,9 +152,9 @@ POST host:port/chaincode
 }
 ```
 
-**Note:** When security is enabled, modify the CLI command and the REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollmentID</b> is passed with the <b>'secureContext'</b> element.
+**Note:** When security is enabled, modify the CLI command and the REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollmentID</b> is passed with the <b>'secureContext'</b> element. If you are enabling security and privacy on the peer process with environment variables, it is important to include these environment variables in the command when executing all subsequent peer operations (e.g. deploy, invoke, or query).
```
Comment (srderson) [inline docs/API/SandboxSetup.md]: Same comment. I think ` may be a better choice than <b>

File: docs/API/SandboxSetup.md [hunk 3]
```diff
@@ -224,9 +224,9 @@ POST host:port/chaincode
 }
 ```
 
-**Note:** When security is enabled, modify the CLI command and REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollmentID</b> is passed with the <b>'secureContext'</b> element.
+**Note:** When security is enabled, modify the CLI command and REST API payload to pass the <b>enrollmentID</b> of a logged in user. To log in a registered user through the CLI or the REST API, follow the instructions in the [note on security functionality](#note-on-security-functionality). On the CLI, the <b>enrollmentID</b> is passed with the <b>-u</b> parameter; in the REST API, the <b>enrollmentID</b> is passed with the <b>'secureContext'</b> element. If you are enabling security and privacy on the peer process with environment variables, it is important to include these environment variables in the command when executing all subsequent peer operations (e.g. deploy, invoke, or query).
```
Comment (srderson) [inline docs/API/SandboxSetup.md]: Same comment. Realize these were <b> before your change. Maybe it can be cleaned up in another PR, but would like to get your opinion.

File: docs/dev-setup/devnet-setup.md [hunk 4]
```diff
@@ -50,29 +50,40 @@ By default, we are using a consensus plugin called `NOOPS`, which doesn't really
...
 ```
 
-You can start up a few more validating peers in the similar manner as you wish. Remember to change the ID.
+Again, the validating peer (enrollID vp1 and enrollSecret vp1_secret) has to be added to membersrvc.yaml file (in fabric/membersrvc).
+
+You can start up a few more validating peers in the similar manner as you wish. Remember to change the peer ID and add the enrollID/enrollSecret to the membersrvc.yaml.
+
+### Enroll/Login a test user (if security is enabled):
+If security is enabled, you must enroll a user with the certificate authority before sending requests. Choose a user that is already registered, i.e. added to the membersrvc.yaml file. Then, execute the command below to log in the user on the target validating peer. `CORE_PEER_ADDRESS` specifies the target validating peer for which the user is to be logged in.
```
Comment (srderson) [inline docs/dev-setup/devnet-setup.md]: Consider making "membersrvc.yaml" a link? [membersrvc.yaml](https://github.com/hyperledger/fabric/blob/master/membersrvc/membersrvc.yaml)

File: docs/dev-setup/devnet-setup.md [hunk 5]
```diff
@@ -50,29 +50,40 @@ By default, we are using a consensus plugin called `NOOPS`, which doesn't really
...
+
+```
+CORE_PEER_ADDRESS=172.17.0.2:30303 ./peer network login jim
+```
+
+**Note:**, the certificate authority allows the enrollID and enrollSecret credentials to be used only *once*. Therefore, login by the same user from any other validating peer will result in an error. Currently, the application layer is responsible for duplicating the crypto material returned from the CA to other peer nodes. If you want to test secure transactions from more than one peer node without replicating the returned key and certificate, you can log in with a different user on other peer nodes.
```
Comment (srderson) [inline docs/dev-setup/devnet-setup.md]: Capitalize 't' in 'the' to be consistent with other Notes

**Comment by srderson** [comment]
Great, nice to see cleanup in docs!

File: docs/API/SandboxSetup.md (hunk 3 above)
Comment (angrbrd) [inline docs/API/SandboxSetup.md]: I have gone through both of the files in this PR to make sure that all instances of <b>xxx</b> have been replaced with `...` where appropriate. About to push new changes.

File: docs/dev-setup/devnet-setup.md (hunk 4 above)
Comment (angrbrd) [inline docs/dev-setup/devnet-setup.md]: Done.

File: docs/dev-setup/devnet-setup.md (hunk 5 above)
Comment (angrbrd) [inline docs/dev-setup/devnet-setup.md]: Done.

**Comment by angrbrd** [comment]
Didn't realize I'm in conflict now... let me rebase.
